### PR TITLE
Lint and format

### PR DESCRIPTION
**Pull request by kayleg**
Ran `rustfmt` over the entire crate.
Applied the `cargo clippy` lint recommendations.

This includes the changes from #20 so that should be merged first.

### CODE REVIEW

**Comment by WillSewell** [comment]
Thanks! I'll hold off merging this.

I'd also like to add a step to the CI process to ensure that the code complies with `rustfmt`. That doesn't necessarily have to be in this PR though.

**Comment by WillSewell** [comment]
@kayleg just a reminder that this will need rebasing

File: README.md [hunk 1]
```diff
@@ -145,11 +145,11 @@ pusher.trigger("test_channel", "my_event", &hash_map).await;
 
 #### Multiple channels
 
-##### `async fn trigger_multi<S: serde::Serialize>(&self, channels: &Vec<&str>, event: &str, payload: S)`
+##### `async fn trigger_multi<S: serde::Serialize>(&self, channels: &[&str], event: &str, payload: S)`
 
 |Argument | Description |
 |:-:|:-:|
-|channels `&Vec<&str>`| A vector of channel names you wish to send an event on. The maximum length is 10.|
+|channels `&[&str]`| A vector of channel names you wish to send an event on. The maximum length is 10.|
```
Comment (WillSewell) [inline README.md:152]: Is this a breaking change?

Comment (kayleg) [reply README.md:152]: No - Since the `Vec` is not mutable this is equivalent and is more idiomatic.

**Comment by WillSewell** [comment]
Thanks!

**Pull request merged**